core/src/machine/mod.rs

```diff
@@ -97,6 +121,29 @@ impl Machine {
         self.cpu.sp -= 2u16;
         self.store_word(self.cpu.sp, word);
     }
+
+    /// Executes the interrupt service routine and returns the number of clocks used for this.
```

Comment:
I think "Executes" is confusing here, because the ISR is not actually executed. I'd say "Calls the interrupt service routine ...".